MapboxNavigation/SpeechSynthesizersController.swift

```diff
@@ -0,0 +1,116 @@
+
+import AVFoundation
+import MapboxDirections
+import MapboxCoreNavigation
+import MapboxSpeech
+
+/// `SpeechSynthesizing`implementation, aggregating other implementations, to allow 'fallback' mechanism.
+/// Can be initialized with array of synthesizers which will be called in order of appearance, untill one of them is capable to vocalize current `SpokenInstruction`
+open class SpeechSynthesizersController: SpeechSynthesizing {
+    
+    // MARK: - Properties
+    
+    public var delegate: SpeechSynthesizingDelegate?
+    
+    public var muted: Bool = false {
+        didSet {
+            speechSynthesizers.forEach { $0.muted = muted }
+        }
+    }
+    public var volume: Float = 1.0 {
+        didSet {
+            speechSynthesizers.forEach { $0.volume = volume }
+        }
+    }
+    public var isSpeaking: Bool {
+        return speechSynthesizers.first(where: { $0.isSpeaking }) != nil
+    }
+    public var locale: Locale = Locale.autoupdatingCurrent {
+        didSet {
+            speechSynthesizers.forEach { $0.locale = locale }
+        }
+    }
+    
+    public  let speechSynthesizers: [SpeechSynthesizing]
+    
+    private var currentLegProgress: RouteLegProgress?
+    
+    // MARK: - Lifecycle
+    
+    public init(_ speechSynthesizers: [SpeechSynthesizing]? = nil, accessToken: String? = nil) {
+        self.speechSynthesizers = speechSynthesizers ?? [
+            MapboxSpeechSynthesizer(accessToken),
+            SystemSpeechSynthesizer()
+        ]
+        
+        self.speechSynthesizers.forEach { $0.delegate = self }
+    }
+    
+    // MARK: - Public Methods
+    
+    public func changedIncomingSpokenInstructions(_ instructions: [SpokenInstruction]) {
+        speechSynthesizers.forEach { $0.changedIncomingSpokenInstructions(instructions) }
+    }
+    
+    public func speak(_ instruction: SpokenInstruction, during legProgress: RouteLegProgress) {        
+        guard let synthesizer = speechSynthesizers.first else {
+            assert(false, "SpeechSynthesizersController has 0 speechSynthesizers")
+            delegate?.speechSynthesizer(self,
+                                        didSpeak: instruction,
+                                        with: nil)
```

Comment:
This case is an error on the part of the developer. I think it’s possible for the application to be built with the `-assert-config` flag that disables assertions, in which case this developer error becomes a silent failure. Depending on the severity or recoverability of the issue, it would be best to either issue a `preconditionFailure(_:file:line:)` or throw an error.